src/network.c

```diff
@@ -129,6 +131,8 @@ gboolean download_file(const gchar *target, const gchar *url, gsize limit, GErro
 	gboolean res = FALSE;
 	GError *ierror = NULL;
 
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```

Comment:
Input checks for `target` and `url` should be added, too.

Reply:
Done.